### PR TITLE
Fix #1045: PropertyBagConverter uses JavaScriptStringEncode

### LINKED ISSUE #1045: PropertyBagConverter uses JavaScriptStringEncode
@boAndron reported this bug:

The `PropertyBagConverter` class uses `System.Web.HttpUtility.JavaScriptStringEncode(serializedValue)` to encode the value of property values if they are strings (line 48). This is inconsistent with the serialization of any other string in the model, because `JavaScriptStringEncode` will replace some control characters with their Unicode value (such as apostrophes to `\u0027`). Also, the encoding is done just the deserialization step (not the serialization step), leading to more inconsistency.

### PR DESCRIPTION
**Pull request by lgolding**
This bug caused string-valued properties to be deserialized incorrectly, so when they were re-serialized, the resulting string did not match the original string. The fix is to use `JsonConvert.ToString` rather than `JavaScriptStringEncode`.

Also:
- Start marking tests with the bug the were introduced to regress, using the XUnit `Trait` attribute.

### CODE REVIEW

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

**Comment by lgolding** [comment]
@boAndron is added to the review. #Closed

**Comment by lgolding** [comment]
@EasyRhinoMSFT is added to the review. #Closed

**Comment by lgolding** [comment]
@Evmaus-MS is added to the review. #Closed

File: src/Sarif.FunctionalTests/PropertyBagConverterTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ""version"": """ + SarifUtilities.SarifFormatVersion + @""",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner""
+      },
+      ""properties"": {
+        ""int"": 42,
+        ""string"": ""'\""\\'""
+      }
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/PropertyBagConverterTests.cs]: @boAndron: This is a minimal repro based on the sample you provided. It fails without the fix, exactly as your sample did, and succeeds with the fix.

File: src/Sarif.FunctionalTests/SarifLogEqualityComparerTests.cs [hunk 2]
```diff
@@ -4,10 +4,11 @@
 using System.IO;
 using FluentAssertions;
 using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.TestUtilities;
 using Newtonsoft.Json;
 using Xunit;
 
-namespace Microsoft.CodeAnalysis.Sarif.UnitTests.Autogenerated
+namespace Microsoft.CodeAnalysis.Sarif.FunctionalTests
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/SarifLogEqualityComparerTests.cs]: >FunctionalTests [](start = 39, length = 15)

Namespace was wrong.

File: src/Sarif.FunctionalTests/PropertyBagConverterTests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  ]
+}";
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/PropertyBagConverterTests.cs]: @michaelcfanning: Yeah, yeah, test framework, yada yada. We'll get there.

File: src/Sarif.FunctionalTests/PropertyBagConverterTests.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            var serializerSettings = new JsonSerializerSettings
+            {
+                ContractResolver = SarifContractResolver.Instance,
+                Formatting = Formatting.Indented,
```
Comment (EasyRhin0) [inline src/Sarif.FunctionalTests/PropertyBagConverterTests.cs]: >, [](start = 48, length = 1)

Nit

**Pull request merged**